drivers/system_pathways/app/models/system_pathways/equity/race_and_ethnicity.rb

```diff
@@ -0,0 +1,117 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module SystemPathways::Equity::RaceAndEthnicity
+  extend ActiveSupport::Concern
+
+  private def race_and_ethnicity_chart_data
+    {
+      chart: 'race_and_ethnicity',
+      config: {
+        size: {
+          height: 800,
+        },
+      },
+      data: race_and_ethnicity_data,
+      table: as_table(race_and_ethnicity_counts, ['Project Type'] + race_ethnicity_combinations.values),
+      link_params: {
+        columns: [[]] + race_ethnicity_combinations.keys.map { |k| ['details[race_ethnicity_combinations][]', k] },
+        rows: [[]] + node_names.map { |k| ['node', k] },
+      },
+    }
+  end
+
+  def race_and_ethnicity_counts
+    @race_and_ethnicity_counts ||= node_names.map do |label|
+      counts = race_ethnicity_combinations.values.map { |m| [m, 0] }.to_h
+      # Get rid of the distinct from node_clients
+      scope = SystemPathways::Enrollment.joins(:client).where(id: node_clients(label).select(:id))
+      race_data = pluck_to_hash(race_columns.except('race_none').map { |k, v| [k, v] }.to_h, scope)
+      race_ethnicity_combinations.each do |key, value|
+        counts[value] = lookups[key]&.call(race_data)
+      end
+      [
+        label,
+        counts,
+      ]
+    end.to_h
+  end
+
+  private def single_race(race_data, race)
+    race_data.select { |client| client[race] && !client['hispanic_latinaeo'] }.
+      map { |client| client.except('id', 'hispanic_latinaeo').values }.
+      count { |m| m.count(true) == 1 }
+  end
+
+  private def single_race_latinaeo(race_data, race)
+    race_data.select { |client| client[race] && client['hispanic_latinaeo'] }.
+      map { |client| client.except('id', 'hispanic_latinaeo').values }.
+      count { |m| m.count(true) == 1 }
+  end
+
+  private def lookups
+    {
+      am_ind_ak_native: ->(race_data) { single_race(race_data, 'am_ind_ak_native') },
+      am_ind_ak_native_hispanic_latinaeo: ->(race_data) { single_race_latinaeo(race_data, 'am_ind_ak_native') },
+      asian: ->(race_data) { single_race(race_data, 'asian') },
+      asian_hispanic_latinaeo: ->(race_data) { single_race_latinaeo(race_data, 'asian') },
+      black_af_american: ->(race_data) { single_race(race_data, 'black_af_american') },
+      black_af_american_hispanic_latinaeo: ->(race_data) { single_race_latinaeo(race_data, 'black_af_american') },
+      hispanic_latinaeo: ->(race_data) do
+        race_data.select { |client| client['hispanic_latinaeo'] }.
+          map { |client| client.except('id').values }.
+          count { |m| m.count(true) == 1 }
+      end,
```

Comment:
I think this needs a comment about what's going on.  I _think_ the `client` isn't actually a client, but some sort of object that has rows with each race key and 0 or 1?